tooling/cli/src/mobile/ios/mod.rs

```diff
@@ -326,70 +326,210 @@ impl From<plist::Value> for PlistKind {
   }
 }
 
-fn merge_plist(src: Vec<PlistKind>, dest: &Path) -> Result<()> {
-  let mut dest_plist = None;
+fn merge_plist(src: Vec<PlistKind>) -> Result<plist::Value> {
+  let mut merged_plist = plist::Dictionary::new();
 
   for plist_kind in src {
     let plist = match plist_kind {
       PlistKind::Path(p) => plist::Value::from_file(p),
       PlistKind::Plist(v) => Ok(v),
     };
     if let Ok(src_plist) = plist {
-      if dest_plist.is_none() {
-        dest_plist.replace(plist::Value::from_file(dest)?);
-      }
-
-      let plist = dest_plist.as_mut().expect("plist not loaded");
-      if let Some(plist) = plist.as_dictionary_mut() {
-        if let Some(dict) = src_plist.into_dictionary() {
-          for (key, value) in dict {
-            plist.insert(key, value);
-          }
+      if let Some(dict) = src_plist.into_dictionary() {
+        for (key, value) in dict {
+          merged_plist.insert(key, value);
         }
       }
     }
   }
 
-  if let Some(dest_plist) = dest_plist {
-    dest_plist.to_file_xml(dest)?;
-  }
-
-  Ok(())
+  Ok(plist::Value::Dictionary(merged_plist))
 }
 
 pub fn signing_from_env() -> Result<(
   Option<tauri_macos_sign::Keychain>,
   Option<tauri_macos_sign::ProvisioningProfile>,
 )> {
-  let keychain = if let (Some(certificate), Some(certificate_password)) = (
+  let keychain = match (
     var_os("IOS_CERTIFICATE"),
     var_os("IOS_CERTIFICATE_PASSWORD"),
   ) {
-    tauri_macos_sign::Keychain::with_certificate(&certificate, &certificate_password).map(Some)?
-  } else {
-    None
+    (Some(certificate), Some(certificate_password)) => {
+      log::info!("Reading iOS certificates from ");
+      tauri_macos_sign::Keychain::with_certificate(&certificate, &certificate_password).map(Some)?
+    }
+    (Some(_), None) => {
+      log::warn!("The IOS_CERTIFICATE environment variable is set but not IOS_CERTIFICATE_PASSWORD. Ignoring the certificate...");
+      None
+    }
+    (None, Some(_)) => {
+      log::warn!("The IOS_CERTIFICATE_PASSWORD environment variable is set but not IOS_CERTIFICATE. Ignoring the certificate...");
```

Comment:
should probably skip this warning, doesn't make sense to check `IOS_CERTIFCATE_PASSWORD `and not `IOS_CERTIFICATE`